src/main/java/crazypants/enderio/conduit/liquid/LiquidConduitRenderer.java

```diff
@@ -113,27 +114,28 @@ public static void renderFluidOutline(CollidableComponent component, FluidStack
         }
     }
 
-    private static Map<CollidableComponent, Map<Fluid, List<CachableRenderStatement>>> cache = new WeakHashMap<CollidableComponent, Map<Fluid, List<CachableRenderStatement>>>();
+    private static final Map<CollidableComponent, Map<Fluid, List<CachableRenderStatement>>> cache = Collections
+            .synchronizedMap(new WeakHashMap<>());
 
     public static List<CachableRenderStatement> computeFluidOutlineToCache(CollidableComponent component, Fluid fluid,
             double scaleFactor, float outlineWidth) {
 
         Map<Fluid, List<CachableRenderStatement>> cache0 = cache.get(component);
         if (cache0 == null) {
-            cache0 = new HashMap<Fluid, List<CachableRenderStatement>>();
+            cache0 = new ConcurrentHashMap<>();
             cache.put(component, cache0);
         }
         List<CachableRenderStatement> data = cache0.get(fluid);
```

Comment:
`cache0` is not updated with whatever concurrent hash map happens to successfully be inserted prior, after the `cache.put` it should be reacquired, because if you don't you have a race condition and you can insert something that will vanish.  There's a similar issue below as well but it doesn't look like there can be data loss in that case, just an inefficiency.  A proper way to insert the above would be a double lock, but the slight efficiency loss doesn't matter in this case.